core/modules/node/node.module

```diff
@@ -2928,7 +2940,37 @@ function _node_query_node_access_alter($query, $type) {
  *   new ones. Defaults to TRUE.
  */
 function node_access_acquire_grants(Node $node, $delete = TRUE) {
-  $grants = module_invoke_all('node_access_records', $node);
+  // We only care about the node if is unpublished. If not, it is
+  // treated just like any other node and we completely ignore it.
+  $grants = array();
+  if ($node->status == 0) {
+    // Unpublished nodes should be viewable to all editors.
+    $grants[] = array(
+      'realm' => 'view_unpublished_content',
+      'gid' => 1,
+      'grant_view' => 1,
+      'grant_update' => 0,
+      'grant_delete' => 0,
+      'priority' => 0,
+    );
+    $grants[] = array(
+      'realm' => 'view_unpublished_' . $node->type . '_content',
+      'gid' => 1,
+      'grant_view' => 1,
+      'grant_update' => 0,
+      'grant_delete' => 0,
+      'priority' => 0,
+    );
+    $grants[] = array(
+      'realm' => 'view_unpublished_author',
+      'gid' => $node->uid,
+      'grant_view' => 1,
+      'grant_update' => 0,
+      'grant_delete' => 0,
+      'priority' => 0,
+    );
+  }
```

Comment:
It looks like this bit of code is functionally changing the way that "view own unpublished" is working. Are the tests changed/updated to reflect this as well?